### ISSUE TITLE
feat(vscode extension): node type gutter markers 

### ISSUE THREAD

**Issue by nokome** [stencila/stencila#2248]
Our editor designs include gutter markers showing node types:

![image](https://github.com/stencila/stencila/assets/1152336/3b16e152-0e6a-4718-98d9-10299de19457)

These were implemented in our CodeMirror based editor and proved useful from a UX perspective for understanding the structure of the document (nesting particularly). Having the same icons and colors in the gutter as in the preview helps understand the relationship between the source code and the preview.

I have done a quick PoC of implementing these in the VSCode extension. It requires on-the-fly generation of SVG images to use in the `gutterIconPath` property of  `TextEditorDecorationType` which is less elegant than using CSS but works.

![image](https://github.com/stencila/stencila/assets/1152336/aa8181be-4749-4d75-ba32-428826adc5f4)

**Comment by jduckles** [comment]
Perhaps, adding "Running" to some kind of gutter visual as well. I've found in testing prompts, that in VSCode, with the preview window open, a prompt doesn't have to be very long for the coordinate where "Running" is placed when a prompt is run is off-screen. In user testing I've seen users hit play 2 and 3 times because there is no visual feedback that the prompt is running when "Running" is off screen.

**Comment by kusalekanayake** [comment]
Initial attempts are looking very similar to the source code version. I haven't added in colours or finished the look but I am feeling that this could be the general approach.
 ![Image](https://github.com/user-attachments/assets/fb010d48-0849-40af-9931-4051021104e6)

**Comment by nokome** [comment]
Personally, I prefer the orginial designs we had for the gutters (screenshot above) - solid vertical bars and no border. Feel less busy to me.

**Comment by kusalekanayake** [comment]
Yeah I personally feel the same. I although I wanted to add a little bit of dimension to make them pop slightly, rather that be flat. ![Image]
I like this approach where there is a slight inner shadow/gradient border. Not sure how  easy it would be to implement though.
(https://github.com/user-attachments/assets/7fe2958d-5083-4709-887c-056174408e14)

**Comment by mike-parkin** [comment]
heres a really rough POC, this is just replacing the 'chip' element with the bars, so all the positioning is the same it just stretches the height of the content slot.

this does not take nesting into account yet. As that will require a bit of thought

![Image](https://github.com/user-attachments/assets/e1e07067-1cab-4c86-9928-cf098c56dca1)

**Comment by nokome** [comment]
That's a great start Mike! For the nesting, the `depth` attribute we added to to all `Entity` components was intended to help here. Can't that be used to simply shift the chips right a few pixels for every unit of depth?

**Comment by nokome** [comment]
Thanks. Ideally, the top of the chip would stretch to the top of the card, IMO.

**Comment by nokome** [comment]
Done
